source('scripts/R/utils.R')
#to do - remove unnecessary calls in utils.R and make this 'source' work
#for now, manually execute the function definition before executing the function calls below


create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 1)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 2)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 3)

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 4)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 5)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 6)

create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "proteomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 7)
create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 8)




create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 9)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 10)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 11)

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 12)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 13)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 14)

create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 15)
create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 16)


create_dim_red_plots_PMH(comparison = "Melanoma_metVsOther", classes = c("Other", "Melanoma_met"),
                         omics_type = "proteomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 17)
create_dim_red_plots_PMH(comparison = "Melanoma_metVsOther", classes = c("Other", "Melanoma_met"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 18)


create_dim_red_plots_PMH(comparison = "Melanoma_metVsOther", classes = c("Other", "Melanoma_met"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 19)
create_dim_red_plots_PMH(comparison = "Melanoma_metVsOther", classes = c("Other", "Melanoma_met"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red/",
                         best_features_file_path = NA,
                         dataset_replace_str = NA,
                         file_name_prefix = 20)




#best biomarker plots
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 1,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 2,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")

create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 3,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")



create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 4,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 5,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")

create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best/",
                         file_name_prefix = 6,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot/")




#following are for testing MET Vs HC biomarker sets on other groups - just to check the expression levels

#below failed to create other biomarker expression since not found in that cohort
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2_test_/qc/dim_red_best/",
                         file_name_prefix = 1,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col_incorrect_rows_added_for_test.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2_test_/qc/boxplot/")

#below failed to create other biomarker expression since not found in that cohort
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2_test_/qc/dim_red_best/",
                         file_name_prefix = 2,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col_incorrect_rows_added_for_test.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2_test_/qc/boxplot/")

#below failed to create other biomarker expression since not found in that cohort
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2_test_/qc/dim_red_best/",
                         file_name_prefix = 4,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col_incorrect_rows_added_for_test.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2_test_/qc/boxplot/")

#only this function call actually created other biomarker expression plot
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2_test_/qc/dim_red_best/",
                         file_name_prefix = 5,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col_incorrect_rows_added_for_test.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2_test_/qc/boxplot/")

# END of
#   following are for testing MET Vs HC biomarker sets on other groups - just to check the expression levels




#rerun to create boxplots with cohort
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best_test_ignore/",
                         file_name_prefix = 1,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot_with_cohort/")

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "proteomics", norm = "quantile_train_param",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = FALSE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best_test_ignore/",
                         file_name_prefix = 2,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_proteomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot_with_cohort/")



create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best_test_ignore/",
                         file_name_prefix = 4,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot_with_cohort/")

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_comparison_set2/qc/dim_red_best_test_ignore/",
                         file_name_prefix = 5,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_combat_compset2_",
                         boxplot_dir_path = "plots_comparison_set2/qc/boxplot_with_cohort/")
#end rerun to create boxplots with cohort




############# new tra data

create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 1,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 2,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 3,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)

create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 4,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 5,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 6,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)

create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 7,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 8,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)




create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/dim_red_best/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/boxplot_best/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 1,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_new_quant_compset2_")
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/dim_red_best/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/boxplot_best/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 2,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_new_quant_compset2_")
create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/dim_red_best/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/boxplot_best/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 3,
                         best_features_file_path = "Data/selected_features/best_features_with_add_col.csv",
                         dataset_replace_str = "GBM_combined_transcriptomic_new_quant_compset2_")




#plots for DE transcripts

create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 1,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 2,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsMET", classes = c("MET", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsMET.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 3,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)


create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 4,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 5,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "PREOPEVsHC", classes = c("HC", "PREOPE"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "combat",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_PREOPEVsHC.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 6,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)



create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "none",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_METVsHC.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_METVsHC.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 7,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
create_dim_red_plots_PMH(comparison = "METVsHC", classes = c("HC", "MET"),
                         omics_type = "transcriptomics", norm = "log_cpm",
                         dim_red = "UMAP",
                         shownames = FALSE,
                         perform_filter = TRUE,
                         batch_effect_correction = "none",
                         plot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/dim_red/",
                         boxplot_dir_path = "plots_RNA_all/PREOPE_MET_HC/DE_qc/boxplot/",
                         data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_METVsHC.csv",
                         validation_data_file_path = "Data/RNA_all/newquant_Nov2023_umi_counts_PREOPE_MET_HC_filter90_de_METVsHC.csv",
                         phenotype_file_path = "Data/transcriptomic_phenotype_PREOPE_MET_HC.txt",
                         file_name_prefix = 8,
                         best_features_file_path = NA,
                         dataset_replace_str = NA)
